src/lib/components/collections/collections-list/ImportCollection.svelte

```diff
@@ -31,20 +31,67 @@
   let isSyntaxError: boolean = false;
   let isResponseLoading: boolean = false;
   let importData: string = "";
-  let importFile: File;
 
-  let fileName = "";
-  function handleFilesSelect(event: CustomEvent) {
-    fileName = event.detail ? event.detail.acceptedFiles[0]?.name : "";
-    importFile = event.detail ? event.detail.acceptedFiles[0] : "";
-  }
   const handleError = () => {
     isSyntaxError = false;
     isDataEmpty = false;
   };
   const handleInputField = (e) => {
     importData = e.target.value;
   };
+  let uploadCollection = {
+    file: {
+      value: [],
+      invalid: false,
+      showFileSizeError: false,
+      showFileTypeError: false,
+    },
+  };
+  const handleLogoInputChange = (
+    e: any,
+    maxSize: number,
+    supportedFileTypes: string[],
+  ) => {
+    if (
+      (e?.target?.files && e?.target?.files[0].size > maxSize * 1024) ||
+      (e?.dataTransfer?.files &&
+        e?.dataTransfer?.files[0].size > maxSize * 1024)
+    ) {
+      uploadCollection.file.showFileSizeError = true;
+      uploadCollection.file.invalid = true;
+      return;
+    }
+    const fileType = `.${(
+      (e?.target?.files && e?.target?.files[0]?.name) ||
+      (e?.dataTransfer?.files && e?.dataTransfer?.files[0]?.name)
+    )
+      .split(".")
+      .pop()
+      .toLowerCase()}`;
+    if (!supportedFileTypes.includes(fileType)) {
+      uploadCollection.file.showFileTypeError = true;
+      uploadCollection.file.invalid = true;
+      return;
+    }
+    uploadCollection.file.showFileSizeError = false;
+    uploadCollection.file.showFileTypeError = false;
+    uploadCollection.file.invalid = false;
+    uploadCollection.file.value =
+      (e?.target?.files && e?.target?.files[0]) ||
+      (e?.dataTransfer?.files && e?.dataTransfer?.files[0]);

```

Comment:
convert this also to const above to fetch here as direct value.

Reply:
resolved